src/Conversion/ONNXToTOSA/ConvertONNXToTOSA.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//====------ ConvertONNXToTOSA.cpp - ONNX dialects to TOSA lowering -------===//
+//
+// Copyright (c) 2022 Arm Limited.
+//
+// =============================================================================
+//
+// This file implements the lowering of frontend operations to the TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+
+#include "src/Dialect/ONNX/DialectBuilder.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "src/Dialect/ONNX/ONNXOpsHelper.hpp"
+#include "src/Pass/Passes.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+// This defines a template to construct ops whose legalizations are
+// specialized.
+template <typename OnnxOpT>
+class ConvertOnnxOp : public OpConversionPattern<OnnxOpT> {
+public:
+  using OpConversionPattern<OnnxOpT>::OpConversionPattern;
+  using OpAdaptor = typename OnnxOpT::Adaptor;
+  LogicalResult matchAndRewrite(OnnxOpT op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override;
+};
+
+template <>
+LogicalResult ConvertOnnxOp<ONNXReluOp>::matchAndRewrite(ONNXReluOp op,
+    OpAdaptor adaptor, ConversionPatternRewriter &rewriter) const {
+  Value input = adaptor.X();
+  auto inputTy = input.getType().cast<TensorType>();
+
+  // Maps to tosa.clamp which has both int and fp limits.
+  int64_t clampMin = 0;
+  Value clampIn = input;
+  if (!inputTy)
+    return op.emitError("Only Tensor types supported in TOSA");
+
+  // Rescale the clampIn for quantized types. TBD
+
+  if (!inputTy.getElementType().isa<mlir::FloatType>()) {
+    return op.emitError(
```

Comment:
Can this semantic constrained be handle earlier than lowering?  Or is this a temporary limitation.  
Either way, suggest to use an assert here.

Reply:
Done